src/metabase/models/action.clj

```diff
@@ -245,14 +246,18 @@
       (if (= (:type action) :implicit)
         (let [model-id        (:model_id action)
               saved-params    (m/index-by :id parameters)
+              action-kind     (:kind action)
               implicit-params (cond->> (get model-id->implicit-parameters model-id)
                                 :always
                                 (map (fn [param] (merge param (get saved-params (:id param)))))
 
-                                (= "row/delete" (:kind action))
+                                (= "row/delete" action-kind)
                                 (filter ::pk?)
 
-                                (contains? #{"row/update" "row/delete"} (:kind action))
+                                (= "row/create" action-kind)
+                                (remove :is-auto-increment)
```

Comment:
We also need to remove non-required PKs from this too, right? Are you going to do that in another PR?

Reply:
I'm not sure I follow, how is PK nullable?

anw, product specifically asks for autoincrements only ([thread](https://metaboat.slack.com/archives/C04FG88HL95/p1678098385553369?thread_ts=1677593986.943879&cid=C04FG88HL95))

Reply:
PKs can't be nullable but as we discussed, PKs can be non-required if there's a default.

Bruno was asking specifically about autoincrements, but you were also right [here](https://metaboat.slack.com/archives/C04FG88HL95/p1678089679191799?thread_ts=1677593986.943879&cid=C04FG88HL95). non-required PKs like `id uuid PRIMARY KEY DEFAULT uuid_generate_v4()` should be excluded too.

Reply:
let's wait for Bruno on that, if we want to do that then we will need to track default as well.

I'll resolve this comment for now because I can't merge without resolving it :(

Reply:
Ah sorry I misunderstood that. yea that makes sense to me too. I'll do that in a follow up